### DISCUSSION TITLE
How to communicate with the robot

### DISCUSSION THREAD

**Discussion by vahidvpa** [openTCS/opentcs#9] [Q&A] [answered]
How the software communicates with the robot and what hardware is used to communicate

**Comment by martingr** [comment, marked as answer]
Communication between openTCS (more specifically the openTCS kernel) and vehicles is done via custom vehicle drivers that implement vehicle-specific communication protocols. How to develop vehicle drivers is described in the [developer's guide](https://www.opentcs.org/docs/5.3/developer/developers-guide/opentcs-developers-guide.html#_developing_vehicle_drivers).

How the communication is implemented exactly is not really relevant as long as the communication hardware exists and can be used from Java software. In many cases, standard Wi-Fi hardware is used.

**Reply by vahidvpa** [reply]
Thank you dear friend
